### PR TITLE
Use log crate instead of custom logging logic

### PR DESCRIPTION
**Pull request by hug-dev**
Trades pretty print with color and such for the simplicity of just using
the log crate. It also allows people to filter the logs with the
RUST_LOG environment variable.
This commit also reviews the output strings of all commands.
Fixes signature ASN.1 structure creation.

Signed-off-by: Hugues de Valon <hugues.devalon@arm.com>

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -40,8 +51,10 @@ fn main() -> std::io::Result<()> {
         client.set_timeout(timeout);
     }
 
-    matches.subcommand.run(client).map_err(|e| {
-        err!("{:?}", e);
-        std::io::Error::new(std::io::ErrorKind::Other, "Executing subcommand failed.")
-    })
+    if let Err(e) = matches.subcommand.run(client) {
+        error!("Subcommand failed: {} ({:?}", e, e);
```
Comment (ionut-arm) [inline src/main.rs:55]:
```suggestion
        error!("Subcommand failed: {} ({:?})", e, e);
```

File: src/subcommands/list_keys.rs [hunk 2]
```diff
@@ -22,11 +23,13 @@ impl ListKeys {
         }
         info!("Available keys:");
         for key in keys {
-            eprint_colored!(Blue, "*");
-            eprint_colored!(Yellow, " '{}'", key.name);
-            eprintln!(
-                " ({}, {} bit {:?})",
-                key.provider_id, key.attributes.bits, key.attributes.key_type
+            println!(
+                "* {} ({}, {:?}, {} bits, algorithm: {:?})",
```
Comment (ionut-arm) [inline src/subcommands/list_keys.rs:27]: Would it make sense to say `permitted algorithm` instead?

File: README.md [hunk 3]
```diff
@@ -1,30 +1,58 @@
...
-For demos and to test the Parsec service, you might want to change the Parsec endpoint location.
-For that, set the `PARSEC_SERVICE_ENDPOINT` environment variable to correction endpoint.
+## Getting started
+
+To compile and list the available commands:
+
+```
+$ cargo build
+$ ./target/debug/parsec-tool --help
+```
```
Comment (ionut-arm) [inline README.md:13]: Maybe you can just show the instruction for installing it? `cargo install --path ./` or smth like that. The `./target/debug...` thing looks kinda unfinished

File: README.md [hunk 4]
```diff
@@ -1,30 +1,58 @@
...
+documentation](https://docs.rs/env_logger/0.8.3/env_logger/index.html) for more information.
+
+## Data format
+
+Unless specified otherwise below, the data format expected by the commands is the same as describe
+in the [Parsec
+Book](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/index.html). The
+`--help` option of commands might give more information about the expected format.
+
+The ECC public keys are formatted using the ASN.1 representation `Ecdsa-Sig-Value` described in [RFC
```
Comment (ionut-arm) [inline README.md:38]:
```suggestion
ECDSA signatures are formatted using the ASN.1 representation `Ecdsa-Sig-Value` described in [RFC
```

Also, it's worth saying that `export_public_key` formats them in PEM

**Review by ionut-arm — COMMENTED** [review]
Thanks for the work to clean this up! 🧹

File: README.md (hunk 3 above)
Comment (hug-dev) [reply README.md:13]: I can replace with `cargo run`? I am a bit afraid of describing an installation method for something as early as `parsec-tool` 😨

File: README.md (hunk 4 above)
Comment (hug-dev) [reply README.md:38]: oops

File: README.md (hunk 3 above)
Comment (ionut-arm) [reply README.md]: Sure, I don't mind - in that case just leave it as is

**Review by ionut-arm — APPROVED** [review]

**Pull request merged**